### PR TITLE
Feat: MemberLeave

### PR DESCRIPTION
**Pull request by shagunmistry**
Remove the member from as many DB as possible
Let me know if I missed any DBs

### CODE REVIEW

File: src/events/MemberLeave.ts [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+export class MemberLeave {
+  constructor(member: GuildMember | PartialGuildMember) {
+    if (member.roles.cache.find((r) => r.name === "Buddy Project 2020")) {
+      RemoveFromBuddyProject(member.id);
+      RemoveFromBirthdays(member.id);
+      RemoveFromGroups(member.id);
+    }
+    member.createDM().then((dm) => {
+      dm.send(
+        `Hi ${member.displayName}! Sorry to hear that you left the server! :( If you would ever like to join back, just let me know! :slight_smile: `
```
Comment (jamie-legg) [inline src/events/MemberLeave.ts:17]: We can't send user's DMs if they have no servers in common. I would opt to remove this functionality altogether.

File: src/events/MemberLeave.ts [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+import { BirthdayRepository } from "../entities/Birthday";
+import { GroupMember, UserGroupRepository } from "../entities/UserGroup";
+import { textLog } from "../common/moderator";
+import { ILike } from "../lib/typeormILIKE";
+
+export class MemberLeave {
+  constructor(member: GuildMember | PartialGuildMember) {
+    if (member.roles.cache.find((r) => r.name === "Buddy Project 2020")) {
+      RemoveFromBuddyProject(member.id);
+      RemoveFromBirthdays(member.id);
```
Comment (jamie-legg) [inline src/events/MemberLeave.ts:12]: Why are we only doing this if they have the buddy project role? Shouldn't birthday and groups be done regardless of if they were in the buddy project or not?

File: src/index.ts [hunk 3]
```diff
@@ -1,37 +1,38 @@
-import {Client, Channel, Emoji, Guild, GuildMember, PartialGuildMember, Message, User, PartialUser, Collection, Role, TextChannel, Snowflake, MessageReaction, Speaking, PartialMessage, Presence, VoiceState,} from 'discord.js';
+import { Client, Channel, Emoji, Guild, GuildMember, PartialGuildMember, Message, User, PartialUser, Collection, Role, TextChannel, Snowflake, MessageReaction, Speaking, PartialMessage, Presence, VoiceState, } from 'discord.js';
```
Comment (jamie-legg) [inline src/index.ts:1]: Unnecessary whitespace
```suggestion
import { Client, Channel, Emoji, Guild, GuildMember, PartialGuildMember, Message, User, PartialUser, Collection, Role, TextChannel, Snowflake, MessageReaction, Speaking, PartialMessage, Presence, VoiceState } from 'discord.js';
```

File: src/events/MemberLeave.ts (hunk 2 above)
Comment (shagunmistry) [reply src/events/MemberLeave.ts]: yep, makes sense
